spec/support/cmap.rb

```diff
@@ -0,0 +1,346 @@
+
+            if err
+              result['error'] = err
+              break
+            end
+
+            # Because we can't determine what order threads will return their connections into the
+            # pool, we add a slight delay between each operation in order to make the scheduling
+            # more consistent.
+            sleep(0.01)
+          end
+
+          result['events'] = subscriber.succeeded_events.reduce([]) do |events, event|
+            event = case event
+                    when Mongo::Monitoring::Event::PoolCreated
+                      {
+                        'type' => 'ConnectionPoolCreated',
+                        'address' => event.address,
+                        'options' => normalize_options(event.options),
+                      }
+                    when Mongo::Monitoring::Event::PoolClosed
+                      {
+                        'type' => 'ConnectionPoolClosed',
+                        'address' => event.address,
+                      }
+                    when Mongo::Monitoring::Event::ConnectionCreated
+                      {
+                        'type' => 'ConnectionCreated',
+                        'connectionId' => event.connection_id,
+                      }
+                    when Mongo::Monitoring::Event::ConnectionReady
+                      {
+                        'type' => 'ConnectionReady',
+                        'connectionId' => event.connection_id,
+                      }
+                    when Mongo::Monitoring::Event::ConnectionClosed
+                      {
+                        'type' => 'ConnectionClosed',
+                        'connectionId' => event.connection_id,
+                        'reason' => event.reason,
+                      }
+                    when Mongo::Monitoring::Event::ConnectionCheckoutStarted
+                      {
+                        'type' => 'ConnectionCheckOutStarted',
+                    }
+                    when Mongo::Monitoring::Event::ConnectionCheckoutFailed
+                      {
+                        'type' => 'ConnectionCheckOutFailed',
+                        'reason' => event.reason,
+                      }
+                   when Mongo::Monitoring::Event::ConnectionCheckedOut
+                      {
+                        'type' => 'ConnectionCheckedOut',
+                        'connectionId' => event.connection_id,
+                      }
+                    when Mongo::Monitoring::Event::ConnectionCheckedIn
+                      {
+                        'type' => 'ConnectionCheckedIn',
+                        'connectionId' => event.connection_id,
+                      }
+                    when Mongo::Monitoring::Event::PoolCleared
+                      {
+                        'type' => 'ConnectionPoolCleared',
+                        'address' => event.address,
+                      }
+                    else
+                      next
```

Comment:
I think it would be helpful to have a base class for all connection pool-related events, so that the user code can also verify that it handled all pool events. Then this case would raise on an event which is (a subclass of) this base class and can ignore non-pool events.